tests/test_matrix_props/test_is_permutation.py

```diff
@@ -0,0 +1,42 @@
+"""Test is_permutation."""
+import numpy as np
+
+from toqito.matrix_props.is_permutation import is_permutation
+
+def test_is_simple_permutation():
+    """Test if matrix is a simple permutation matrix."""
+    mat = np.array([[1, 0, 0], [0, 0, 1], [0, 1, 0]])
+    np.testing.assert_equal(is_permutation(mat), True)
+
+def test_is_not_square():
+    """Test non-square matrix and thus will fail."""
+    mat = np.array([[1, 0, 0 ], [1, 0, 0], [0, 1, 0], [0,0,1]])
+    np.testing.assert_equal(is_permutation(mat), False)
+
+def test_is_not_permutation():
+    """Test non-permutation matrix with one element being 9 !in set (0,1)."""
+    mat = np.array([[0, 0, 0, 9], [1, 0, 0, 0], [0, 1, 0, 0], [0,0,1,0]])
+    np.testing.assert_equal(is_permutation(mat), False)
+
+def test_is_not_permutation_row():
+    """Test non-permutation matrix where the row check will fail."""
+    mat = np.array([[0, 1, 1,0], [1, 0, 0, 0], [0, 0, 0, 1], [0,0,1,0]])
+    np.testing.assert_equal(is_permutation(mat), False)
+
+def test_is_not_permutation_also():

```

Comment:
This test title is not descriptive as to what is being tested. I would recommend changing it to something less arbitrary and more specific.

Reply:
fixing... not easy to describe these examples, however inventing English syntax based on the math.  Hopefully it is easy to understand.